bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdConfigFileChecker.java

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Bachmann electronic GmbH and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.internal.clangd;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.eclipse.cdt.lsp.internal.clangd.editor.ClangdPlugin;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Checks the <code>.clangd</code> file for syntax errors and notifies the user via error markers in the file and Problems view.
+ */
+public class ClangdConfigFileChecker {
+	public static final String CLANGD_MARKER = ClangdPlugin.PLUGIN_ID + ".config.marker"; //$NON-NLS-1$
+	private final Pattern pattern = Pattern.compile(".*line (\\d+), column (\\d+).*"); //$NON-NLS-1$
+	private boolean temporaryLoadedFile = false;
+
+	/**
+	 * Checks if the .clangd file contains valid yaml syntax. Adds error marker to the file if not.
+	 * @param configFile
+	 */
+	public void checkConfigFile(IFile configFile) {
+		Yaml yaml = new Yaml();
+		try (var inputStream = configFile.getContents()) {
```

Comment:
I think there is a mismatch here as this is reading the disk contents of the file, whereas later in processing the error you are using the text file buffer. If I read this properly it means you can have a mismatch between what is on disk and what is in a dirty editor.

Reply:
I think this is no problem. My `IResourceChangeListener` in `ClangdConfigFileMonitor`  gets called when the file gets saved. In my first attempt, I tried to do the check in the `IResourceDeltaVisitor`, This fails, when I wanted to add a marker to the file, because there was write lock on it. That's why I schedule the check in the `checkJob` which is a `WorkspaceJob`. I'am assuming, when the `ClangdConfigFileChecker.checkConfigFile(IFile)` gets called, the file is saved and the file document should be equal to the buffer.

Reply:
But it would be safer and easier when I am working on the buffer I think

Reply:
> My IResourceChangeListener in ClangdConfigFileMonitor gets called when the file gets saved.

There is a race condition, becuase by the time the resource change listener is run and the file is checked, the buffer can become dirty again. But I understand your point.

> But it would be safer and easier when I am working on the buffer I think

+1